README.md

```diff
@@ -173,6 +173,8 @@ Community Maintained Cars and Features
 | Subaru    | Crosstrek 2018-19             | EyeSight          | Stock            | 0mph               | 0mph         |
 | Subaru    | Forester 2019-20              | EyeSight          | Stock            | 0mph               | 0mph         |
 | Subaru    | Impreza 2017-19               | EyeSight          | Stock            | 0mph               | 0mph         |
+| Tesla | Model S 2014-2016 | AutoPilot HW1 | Stock | 0mph | 0mph |
```

Comment:
Spacing here

Reply:
@geohot @robbederks  So tesla has a VERY low threshold for CAN errors... it takes two mismatches for CAN counter to disable steering and ACC cababilities. 

If you look in https://github.com/boggyver/panda/tree/tesla_ap2_fwd you will see a little different approach to block and regenerate. 

I created a set of functions in https://github.com/BogGyver/panda/blob/tesla_ap2_fwd/board/safety/safety_forwards.h (ignore the location for now) that still allows us to generate the message on the Comma2 but it is checked, packed with the official counter and then sent via the fwd hook. All the messages that need to follow this protocol are declared and checked. There are safety checks for timers. If you want to send the message as is just set the counter to 0, otherwise set counter to 1. 

I can explain more if needed but with this approach the counter/frequency is preserved and the message is still generated in the Comma2 side (except of course counter and checkum). The same set of checks can be done in the TX side before the data is accepted. This preserves your logic and existing code (except computing the checksum and counter) on all existing functions to generate messages. 

This safety file implements that method as an example (and i have no more EPAS issues due to counters). https://github.com/BogGyver/panda/blob/tesla_ap2_fwd/board/safety/safety_tesla.h